File: solutions/011.Container_With_Most_Water/AC_two_points_n.java
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_two_points_n.java
*  Create Date: 2015-03-02 22:48:31
*  Descripton:   
*/

import java.util.Scanner;

public class Solution {

    public int maxArea(int[] height) {
        int lpoint = 0, rpoint = height.length - 1;
        int area = 0;
        while (lpoint < rpoint) {
            area = Math.max(area, Math.min(height[lpoint], height[rpoint]) *
                    (rpoint - lpoint));
            if (height[lpoint] > height[rpoint])
                rpoint--;
            else
                lpoint++;
        }
        return area;
    }

    // debug
    public static void main(String[] args) {
        Scanner cin = new Scanner(System.in);
        Solution s = new Solution();
        int[] input = {1, 2, 3, 1};
        System.out.println("no case");
    }
}

File: solutions/011.Container_With_Most_Water/AC_two_points_n.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_two_points_n.cpp
*  Create Date: 2014-12-31 11:13:00
*  Descripton:  left point and right point 
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    int maxArea(vector<int> &height) {
        int lpoint = 0, rpoint = height.size() - 1;
        int area = 0;
        while (lpoint < rpoint) {
            area = max(area, min(height[lpoint], height[rpoint]) *
                    (rpoint - lpoint));
            if (height[lpoint] > height[rpoint])
                rpoint--;
            else
                lpoint++;
        }
        return area;
    }
};

int main() {

    return 0;
}

